### PR TITLE
Update to new parent POM

### PR DESCRIPTION
**Pull request by jglick**
@reviewbybees esp. @andresrc

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -284,16 +278,22 @@ THE SOFTWARE.
       <groupId>org.apache.maven.wagon</groupId>
       <artifactId>wagon-http</artifactId>
       <version>${wagonVersion}</version>
+      <exclusions>
+        <exclusion>
+          <groupId>commons-io</groupId>
+          <artifactId>commons-io</artifactId>
```
Comment (jglick) [inline pom.xml]: Old version, clashed with a dependency needed by tests.

File: pom.xml [hunk 2]
```diff
@@ -284,16 +278,22 @@ THE SOFTWARE.
...
       <groupId>org.apache.httpcomponents</groupId>
       <artifactId>httpclient</artifactId>
-      <version>4.2.5</version>
+      <version>4.5.1</version>
     </dependency>
     <dependency>
       <groupId>org.apache.httpcomponents</groupId>
       <artifactId>httpcore</artifactId>
-      <version>4.2.4</version>
+      <version>4.4.3</version>
```
Comment (jglick) [inline pom.xml]: Also clashed with new HtmlUnit.

File: pom.xml [hunk 3]
```diff
@@ -408,17 +405,35 @@ THE SOFTWARE.
       <artifactId>mockito-core</artifactId>
       <version>1.8.5</version>
       <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>org.hamcrest</groupId>
+          <artifactId>hamcrest-core</artifactId>
```
Comment (jglick) [inline pom.xml]: Trashy old version.

File: pom.xml [hunk 4]
```diff
@@ -408,17 +405,35 @@ THE SOFTWARE.
...
     <dependency>
       <groupId>org.powermock</groupId>
       <artifactId>powermock-api-mockito</artifactId>
-      <version>1.4.10</version>
+      <version>1.6.4</version>
+      <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>org.mockito</groupId>
+          <artifactId>mockito-all</artifactId>
```
Comment (jglick) [inline pom.xml]: Bundled Hamcrest in an incompatible version.

File: pom.xml [hunk 5]
```diff
@@ -408,17 +405,35 @@ THE SOFTWARE.
...
+          <groupId>org.hamcrest</groupId>
+          <artifactId>hamcrest-core</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>org.powermock</groupId>
       <artifactId>powermock-module-junit4</artifactId>
-      <version>1.4.10</version>
+      <version>1.6.4</version>
```
Comment (jglick) [inline pom.xml]: Was not compatible with Java 8.

File: pom.xml [hunk 6]
```diff
@@ -442,62 +457,8 @@ THE SOFTWARE.
...
-      </plugins>
-    </pluginManagement>
     <plugins>
       <plugin>
-        <groupId>org.jenkins-ci.tools</groupId>
-        <artifactId>maven-hpi-plugin</artifactId>
-        <!-- version specified in grandparent pom -->
-        <extensions>true</extensions>
-        <configuration>
-          <showDeprecation>true</showDeprecation>
```
Comment (jglick) [inline pom.xml]: Whatever this was, it is gone now.

File: pom.xml [hunk 7]
```diff
@@ -442,62 +457,8 @@ THE SOFTWARE.
...
-          <disabledTestInjection>true</disabledTestInjection>
```
Comment (jglick) [inline pom.xml]: Silly to disable `InjectedTest` but then add `MavenJellyTest` which does a subset of that.

File: pom.xml [hunk 8]
```diff
@@ -442,62 +457,8 @@ THE SOFTWARE.
...
-            </configuration>
-          </execution>
-        </executions>
-      </plugin>
-      <plugin><!-- generate licenses.xml -->
-        <groupId>com.cloudbees</groupId>
-        <artifactId>maven-license-plugin</artifactId>
-        <!-- version specified in grandparent pom -->
-        <configuration>
-          <generateLicenseXml>target/${project.artifactId}/WEB-INF/licenses.xml</generateLicenseXml>
```
Comment (jglick) [inline pom.xml]: Who knows why any of these overrides were here. Probably old, perhaps dating to when this plugin was hosted inside `jenkinsci/jenkins`.

File: src/main/java/hudson/maven/AbstractMavenBuilder.java [hunk 9]
```diff
@@ -127,7 +127,7 @@ protected String formatArgs(List<String> args) {
      * Ignores {@link #systemProps jenkins environment variables} with empty keys.
      * @throws IllegalArgumentException if a {@link #systemProps jenkins environment variable} has null value
      *      as it blows up Maven.
-     * @see http://jenkins.361315.n4.nabble.com/Upgrade-to-1-424-broke-our-Maven-builds-due-to-empty-system-property-key-td3726460.html
+     * @see <a href="http://jenkins.361315.n4.nabble.com/Upgrade-to-1-424-broke-our-Maven-builds-due-to-empty-system-property-key-td3726460.html">Upgrade to 1.424 broke our Maven builds due to empty system property key</a>
```
Comment (jglick) [inline src/main/java/hudson/maven/AbstractMavenBuilder.java]: JDK 8 `javadoc` is stricter.

File: src/test/java/hudson/maven/CaseResultTest.java [hunk 10]
```diff
@@ -44,8 +45,8 @@
      */
     @Bug(4257)
     @Test public void mavenErrorMsgAndStacktraceRender() throws Exception {
-        j.configureDefaultMaven();
-        MavenModuleSet m = j.createMavenProject("maven-render-test");
+        ToolInstallations.configureMaven3();
```
Comment (jglick) [inline src/test/java/hudson/maven/CaseResultTest.java]: The FindBugs plugin in this sample did not run on Java 8, so I had to update the plugin, but then the new version did not run on Maven 2, so I had to update that too.

File: src/test/java/hudson/maven/Maven31xBuildTest.java [hunk 11]
```diff
@@ -49,6 +34,9 @@
     public MavenInstallation configureMaven3x()
         throws Exception
     {
-        return configureMaven31();
+        MavenInstallation mvn = ToolInstallations.configureDefaultMaven("apache-maven-3.1.0", MavenInstallation.MAVEN_30);
+        MavenInstallation m3 = new MavenInstallation("apache-maven-3.1.0", mvn.getHome(), NO_PROPERTIES);
+        jenkins.getDescriptorByType(Maven.DescriptorImpl.class).setInstallations(m3);
```
Comment (jglick) [inline src/test/java/hudson/maven/Maven31xBuildTest.java]: `ToolInstallations` did not get `configureMaven31`. Probably only used in this plugin anyway.

File: src/test/java/hudson/maven/MavenBuildTest.java [hunk 12]
```diff
@@ -74,8 +70,8 @@ public void testMavenWorkspaceExists() throws Exception {
     @Bug(4177)
     @Test
     public void testTestFailureInEarlyTaskSegment() throws Exception {
-        j.configureDefaultMaven();
-        MavenModuleSet m = j.createMavenProject();
+        ToolInstallations.configureMaven3();
```
Comment (jglick) [inline src/test/java/hudson/maven/MavenBuildTest.java]: as in `CaseResultTest`

File: src/test/java/hudson/maven/MavenJellyTest.java [hunk 13]
```diff
@@ -1,40 +0,0 @@
...
-import junit.framework.Test;
-import junit.framework.TestCase;
-import org.jvnet.hudson.test.JellyTestSuiteBuilder;
-
-/**
- * Runs Jelly checks on maven-plugin.
- *
- * @author Andrew Bayer
- */
-public class MavenJellyTest extends TestCase {
```
Comment (jglick) [inline src/test/java/hudson/maven/MavenJellyTest.java]: See note in `pom.xml`.

File: src/test/java/hudson/maven/MavenSnapshotTriggerTest.java [hunk 14]
```diff
@@ -1,7 +1,7 @@
 package hudson.maven;
 
 import static org.junit.Assert.assertThat;
-import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.Matchers.hasItems;
```
Comment (jglick) [inline src/test/java/hudson/maven/MavenSnapshotTriggerTest.java]: Hamcrest update IIRC.

File: src/test/java/hudson/maven/RedeployPublisherTest.java [hunk 15]
```diff
@@ -242,8 +244,8 @@ public boolean accept( File dir, String name )
     @Bug(3773)
     @Test
     public void testDeployUnstable() throws Exception {
-        j.configureDefaultMaven();
-        MavenModuleSet m2 = j.createMavenProject();
+        ToolInstallations.configureMaven3();
```
Comment (jglick) [inline src/test/java/hudson/maven/RedeployPublisherTest.java]: as in `CaseResultTest`

File: src/test/java/hudson/maven/RedeployPublisherTest.java [hunk 16]
```diff
@@ -253,14 +255,15 @@ public void testDeployUnstable() throws Exception {
         j.assertBuildStatus(Result.UNSTABLE, b);
 
         assertTrue("Artifact should have been published even when the build is unstable",
-                   new File(repo,"test/test/1.0-SNAPSHOT/test-1.0-SNAPSHOT.jar").exists());
+            // exact filename unpreductable in M3, e.g. test-1.0-20160317.213607-1.jar
```
Comment (jglick) [inline src/test/java/hudson/maven/RedeployPublisherTest.java]: side effect of M2 → M3

**Comment by andresrc** [comment]
:bee:

**Comment by jglick** [comment]
Increasing [build timeout](https://jenkins.ci.cloudbees.com/job/plugins/job/maven-plugin/configure) and trying again.

**Comment by andresrc** [comment]
:bee:

**Comment by aheritier** [comment]
274 tests more ??
https://jenkins.ci.cloudbees.com/job/plugins/job/maven-plugin/org.jenkins-ci.main$maven-plugin/191/testReport/
and 1 failure
https://jenkins.ci.cloudbees.com/job/plugins/job/maven-plugin/org.jenkins-ci.main$maven-plugin/192/testReport/hudson.maven/MavenProjectTest/testSiteBuild/

:bug: sorry

**Comment by jglick** [comment]
> 274 tests more

Yes, this is a benefit of using the standard `InjectedTest`: [new coverage](https://jenkins.ci.cloudbees.com/job/plugins/job/maven-plugin/org.jenkins-ci.main%24maven-plugin/191/testReport/org.jvnet.hudson.test/PropertiesTestSuite$PropertiesTest/).

> and 1 failure

[That](https://jenkins.ci.cloudbees.com/job/plugins/job/maven-plugin/org.jenkins-ci.main$maven-plugin/192/testReport/junit/hudson.maven/MavenProjectTest/testSiteBuild/) is a `NoSuchMethodError` which seems to have been introduced by using the new Jetty. I will investigate.

**Comment by aheritier** [comment]
Nice finding for the servlet-api versions conflict. :bee:

**Pull request merged**